binaryparse.nim

```diff
@@ -2,14 +2,18 @@
 ## generated reads from a Stream and returns a tuple with each named field.
 ## The general format the macro takes is:
 ##
-## ``[type]<size>: <name>[options]``
+## ``[endian][type]<size>: <name>[options]``
 ##
 ## Where optional fields are in [] brackets and required fields are in <>
 ## brackets. Each field has separate meanings, as described in the table below:
 ##
 ## ========== ==================================================================
 ## Name       Description
 ## ---------- ------------------------------------------------------------------
+## endian     This controls whether the bytes will be read with big or little
+##            endian convention. Possible values are ``le`` for little endian
+##            and ``be`` for big endian. If not specified the system.cpuEndian
+##            is used.
```

Comment:
This should default to big-endian as that was the only thing previously supported. Most machines are little-endian, but many formats are big-endian (probably because it's the typical endianess of most networks).

Reply:
Fair enough